### PR TITLE
Bump soroban-env-host versions

### PR DESCRIPTION
**Pull request by graydon**
This just picks up recent changes in the soroban env (specifically the newest XDR as of today, including removal of ScSpecTypeSet)

### CODE REVIEW

**Comment by sisuresh** [comment]
@graydon looks like there's a conflict to fix.

**Comment by graydon** [comment]
I've fixed the conflict but there were changes required in the refundable fee tests that I've tried all evening to fix and I simply can't understand; I'm at the point of just randomly changing numbers up and down to make them work and I can't even get that working -- changing one will cause two to move in opposite directions in a way I can't comprehend.

@dmkozh could you please pick this up and complete it? I isolated the fee-related changes to the second commit here if you want to just revert it and try again, I won't mind. Whatever you can get to work.

**Comment by dmkozh** [comment]
r+ 0cb9a31

File: src/rust/Cargo.toml [hunk 1]
```diff
@@ -52,11 +52,11 @@ optional = true
 version = "0.0.17"
 git = "https://github.com/stellar/rs-soroban-env"
 package = "soroban-env-host"
-rev = "c2e1c21cf8d44db23a159090e3cbaab741860295"
+rev = "f5c6c0f42b4094481b7f32ecac825ce27c7114e0"
```
Comment (2opremio) [inline src/rust/Cargo.toml:55]: Why not `5dc7547a4b15028787d57a849a7684802d899fc7` like the rest?

Comment (sisuresh) [reply src/rust/Cargo.toml:55]: This is supposed to point to the "previous" host version (the comment has an explanation).

Comment (2opremio) [reply src/rust/Cargo.toml:55]: ah, sorry

**Pull request merged**